### DISCUSSION TITLE
Attackmulset, projcanceltime and suggestion.

### DISCUSSION THREAD

**Discussion by Orochikyocr** [ikemen-engine/Ikemen-GO#856] [Feature Request]
Issue 1
Is Attackmulset still broken?
ignorehitpause if combocount > 0 && map(samemovement) > 0 { 
    AttackMulSet{value:.4}
}
This does not work.

ignorehitpause if combocount > 0 && map(samemovement) > 0 { 
    defencemulset{value:2; redirectid:enemynear,id}
}
This does.

Issue 2
It seems projcanceltime is not working, with or without projID, this is used by some Winmugen chars to return fireballs.
[state -3, debug output]
type = printtoconsole
trigger1 = teamside = 1
text= "Numproj:%d /  Numproj100:%d /canceltimeproj: %d"
params = numproj, numprojid(100) ,ProjCancelTime(100)
ignorehitpause = 1 

Suggestion
Also, I was thinking of a new assertspecial that could disable character from receiving power.
asssertspecial{flag:nopower;}
So the player could not receive power from poweradd and getpower if conditions are met.

**Comment by K4thos** [comment]
can't reproduce issue 1 and 2 on latest github build, so I'm removing bug label. If those problems still exists please prepare simple test case with kfm that showcases it.

**Reply by djonskin** [reply]
[kfmp.zip](https://github.com/user-attachments/files/18549296/kfmp.zip)
I don't know if it counts as a simple enough test case, but I edited this kfm to demonstrate a projectile and at least with simple purpose, ProjCancelTime. In his -3 statedef, I want them to play an alt noise when it detects and cancels by a fellow projectile. Load kfmp for P1 & P2, then Crouch for both players, hit X+Y for both players. Have their own projectile cancel at its first tick, hear the alt noise.
EDIT: pots fixed ProjCancelTime afterward, many thanks! [Pull 2288](https://github.com/ikemen-engine/Ikemen-GO/pull/2288)

**Comment by K4thos** [comment]
@djonskin, after being converted to a discussion, this ticket is meant to be solely about the nopower assertspecial feature request.

If this is a bug report, please create a new ticket on the issues page with these more detailed reproduction steps. Otherwise, it risks being overlooked unless someone begins working on it immediately.

**Comment by potsmugen** [comment]
I experimented with a "nopower" thing in the past in the form of PowerMulSet, but the problem with that is that it also affects power _spending_. Needs a more elaborate implementation if any. Like having a parameter to affect or not power loss. But usually when a sctrl has a lot of "ifs" is when I start thinking it's better approached via char code.
